presto-main/src/main/java/com/facebook/presto/execution/scheduler/ConsistentHashingNodeProvider.java

```diff
@@ -44,27 +49,49 @@ static ConsistentHashingNodeProvider create(Collection<InternalNode> nodes, int
                 activeNodesByConsistentHashing.put(murmur3_32().hashString(format("%s%d", node.getNodeIdentifier(), i), UTF_8).asInt(), node);
             }
         }
-        return new ConsistentHashingNodeProvider(activeNodesByConsistentHashing);
+        return new ConsistentHashingNodeProvider(activeNodesByConsistentHashing, nodes.size());
     }
 
-    public ConsistentHashingNodeProvider(NavigableMap<Integer, InternalNode> candidates)
+    private ConsistentHashingNodeProvider(NavigableMap<Integer, InternalNode> candidates, int nodeCount)
     {
         this.candidates = requireNonNull(candidates, "candidates is null");
+        this.nodeCount = nodeCount;
     }
 
     @Override
     public List<HostAddress> get(String key, int count)
     {
+        checkArgument(count <= nodeCount, format("There are only %d nodes available, cannot provide %d nodes", nodeCount, count));
```

Comment:
would this check make it difficult to test locally since usually we just start one node on my laptop?

Reply:
Well, in that case it doesn't make sense to return 2 candidates. We should fix that logic then?

Reply:
@yuanzhanhku  What do you think? I didn't like the hard coded "2" anyways, so making that configurable might be a good idea? Or just instead of throwing, we can just use `nodeCount` instead of `count`, which should work.

Reply:
synced offline with @yuanzhanhku  will return `nodeCount` number of nodes if count is bigger.